nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/api/AbstractSession.java

```diff
@@ -1010,8 +1010,15 @@ public DocumentRef getParentDocumentRef(DocumentRef docRef) {
 
     @Override
     public DocumentModel getParentDocument(DocumentRef docRef) {
-        Document doc = resolveReference(docRef);
-        Document parentDoc = doc.getParent();
+        Document parentDoc;
+        if (docRef != null && docRef.type() == DocumentRef.PATH) {
+            String targetPath = ((String) docRef.reference());
+            String parentPath = targetPath.substring(0, targetPath.lastIndexOf("/"));
```

Comment:
You should use `'/'` here, as the Sonar rule states: "Put single-quotes around '/' to use the faster "lastIndexOf(char)"

Reply:
I think we should handle the borderline cases here:
- "/doc" -> `parentPath` should be `"/"`
- "/" (root document) -> in the end, `parentDoc` should be `null`

If you could write a unit test focused on `CoreSession#getParentDocument` to validate your changes and these borderline cases. Probably in `TestSQLRepositoryAPI`, though the class name is not very relevant.

Reply:
Yes all this new code should be unit tested from `TestSQLRepositoryAPI`, the fact that there are extra tests in rest-api is not enough.

Reply:
(@ThierryCasanova if you don't know, `TestSQLRepositoryAPI` is a very old name, understand it as if it was named `TestAbstractSession`.) (And also `AbstractSession` should be understood as `CoreSessionImpl` really...)